### ISSUE TITLE
Content bug: empty Subnav sections on many pages

### ISSUE THREAD

**Issue by foolip** [mdn/content#3025]
## What page(s) did you find the problem on?

A non-exhaustive list:

https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles#Subnav
https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_Grid_Layout/Subgrid#subnav (the one I first noticed)
https://developer.mozilla.org/en-US/docs/Web/XPath#subnav
https://developer.mozilla.org/en-US/docs/MDN/At_ten#Subnav

## What is the problem?

These sections are often empty, but not always. It seems like these are the remnants of some previous system where they weren't visible, probably being extracting into a sidebar or menu.

## What did you expect to see?

The empty sections should be removed. But it's not clear how to do that, because they're often followed by other content, the Quick links, and sometimes the Subnav is in a `<div class="hidden">`. What's the right pattern?

**Comment by hamishwillee** [comment]
Almost certainly

**Comment by hamishwillee** [comment]
Almost certainly. There are 13 of these in the docs. Most of them looks like this (a couple are wrapped in a `<div class="hidden">`)
```
<h6 id="Subnav">Subnav</h6>

<section id="Quick_links">

...
```
The quick links are shown the sidebar. I suspect that the subnav was previously a way of highlighting the stuff that went in the sidebar. Removing it appears to make no difference.

Action should probably be to to turn the subnav heading into a comment explaining that the following div will be displayed in the sidebar. 

@peterbe Does that seem reasonable to you. Perhaps also make sure that things like the id "quicklinks" are also documented?

**Comment by peterbe** [comment]
Non-exhaustively, I think there are two hacks going on: 

### 1. E.g. https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles#Subnav

These pages don't have a sidebar KS macro. i.e. there's no `<div>{{CSSRef}}</div>` for example. 
The Yari build will go and extract that `$('#Quick_links')` and remove it from the body and make it the sidebar. 

### 2. E.g. http://localhost:3000/en-US/docs/Web/CSS/CSS_Grid_Layout/Subgrid#subnav

Here the document has a sidebar macro. 
And then it has what looks like a hardcoded big `<section id="Quick_links"> ...` block. 
So these pages get their sidebar from the KS macro (e.g. `<p>{{CSSRef}}</p>`)

--

Basically, you can't have (case insensitively) `document.querySelectorAll('#quick_links')` in two places. If you want to have a Subnav section, you can't use `<section id="Quick_links">` because it'll be sucked up to become the sidebar. 
And if you have a KS macro for the sidebar, *and* a hardcoded `<section id="Quick_links">` in the content, that latter one will be removed. 

In conclusion, every page that falls into the two example categories above should probably have its "Subnav" section removed. If you want to have a section with more links within (or near bottom of) the document, it should basically not use `id="Quick_links"`. 

Does that help @hamishwillee ? 
Would you be able to clean it up based on that?

**Comment by hamishwillee** [comment]
@peterbe Thanks. The question then becomes "if the page is trying to set the sidebar in two places, which SHOULD we use"? My _assumption_ is that by default if an appropriate macro like `{{CSSRef}}` has been applied to the page then a manually defined subnav should be removed. I'll do it that way anyway and we can discuss.

**Comment by peterbe** [comment]
I think it automatically gets done in the right order. 
So if you have:

```html
FRONT_MATTER
---

{{CSSRef}}

<p>Bla</p>

<section id="Quick_links">THIS WILL BE DISCARDED AND LOST</section>
```
It'll keep the first one (produced by executing the `CSSRef` macro. The second one will be entirely deleted. 
Now, that's just how it currently works. We *could* change it to something like this:

```
for (const element in querySelectorAll('#quick_links')) {
  doc.sidebarHTML = $.html(element)
  $(delete).remove();
  break;
}
```

But it doesn't seem like in any of the pages I randomly clicked, that it'd be worth keeping it inside the content. You could still use the `ListSubpagesForSidebar` macro I suppose.

**Comment by hamishwillee** [comment]
Thanks @peterbe

The only reason I questioned which is "best" to keep was the discussions we've had around trying to not use macros any more "in general". I much prefer macros for auto sidebar generation though. The only things I might change about the macro approach is:
- naming them such that it is clear what they affect the sidebar.
- provide a link to the source for the sidebar when the content is rendered during development (make it easier for me as an author to find and edit the sidebar definition.

In any case, the actual Subnav heading never gets pulled into the sidebar and is just a label. It should be removed - or possibly turned into a comment if the quicklinks sections remain.

**Comment by hamishwillee** [comment]
Fixed in #3165 - this also replaces quicklinks with CSS ref macro where relevant.

**Comment by foolip** [comment]
Thanks @hamishwillee!
